src/test/java/org/openrewrite/java/spring/internal/GeneratePropertiesMigratorConfiguration.java

```diff
@@ -78,69 +77,116 @@ public static void main(String[] args) throws IOException {
+                }
+                var recipePath = Paths.get("src/main/resources/META-INF/rewrite/spring-boot-%s%s-properties.yml".formatted(majorMinor[0], majorMinor[1]));
+                writeFileHeader(majorMinor, recipePath);
+                generateReplacementRecipes(deprecations, recipePath);
+                generateCommentRecipesForDeprecations(deprecations, recipePath);
             }
         }
     }
+
+    private static List<SpringConfigurationMetadata.ConfigurationProperty> getDeprecations(ScanResult scanResult, HashSet<Object> alreadyDefined) {
+        ResourceList resources = scanResult.getResourcesMatchingWildcard("**/*spring-configuration-metadata.json");
+        return resources.stream()
+          .flatMap(res -> {
+              try (InputStream inputStream = res.open()) {
+                  var metadata = objectMapper.readValue(inputStream, SpringConfigurationMetadata.class);
+                  return metadata.properties().stream()
+                    .filter(p -> p.deprecation() != null)
+                    .filter(p -> alreadyDefined.add(getPropertyKey(p)));
+              } catch (IOException e) {
+                  throw new UncheckedIOException(e);
+              }
+          })
+          .toList();
+    }
+
+    private static void generateReplacementRecipes(List<SpringConfigurationMetadata.ConfigurationProperty> properties, Path recipePath) throws IOException {
+        var replacements = properties.stream()
+          .filter(d -> (d.deprecation() != null) && d.deprecation().replacement() != null)
+          .sorted(Comparator.comparing(SpringConfigurationMetadata.ConfigurationProperty::name))
+          .toList();
+
+        if (!replacements.isEmpty()) {
+            Files.writeString(recipePath, replacements.stream()
+                .map(r -> """
+                    - org.openrewrite.java.spring.ChangeSpringPropertyKey:
+                        oldPropertyKey: %s
+                        newPropertyKey: %s
+                  """.formatted(
+                  r.name(), requireNonNull(r.deprecation()).replacement())
+                )
+                .collect(joining("", "\n", "\n")),
+              StandardOpenOption.APPEND);
+        }
+    }
+
+    private static void generateCommentRecipesForDeprecations(List<SpringConfigurationMetadata.ConfigurationProperty> properties, Path recipePath) throws IOException {
+        var deprecationsWithoutReplacement = properties.stream()
+          .filter(d -> (d.deprecation() != null) && d.deprecation().replacement() == null)
+          .sorted(Comparator.comparing(SpringConfigurationMetadata.ConfigurationProperty::name))
+          .toList();
+
+        if (!deprecationsWithoutReplacement.isEmpty()) {
+            Files.writeString(recipePath, """
+                      - org.openrewrite.java.spring.InlineCommentSpringProperties:
+                          comment: "This property is deprecated and will be removed in future Spring Boot versions"
```

Comment:
Wasn't there a description for some specific properties that we could use here instead?
Might provide more of an actionable next step in the migration, as opposed to just remove.

Reply:
yes, that would be nice. In this case separate recipe with individual comment per property will be generated. Will check if such kind of information is available in metadata